shared/src/pool_aggregating.rs

```diff
@@ -0,0 +1,68 @@
+use crate::amm_pair_provider::{AmmPairProvider, SushiswapPairProvider, UniswapPairProvider};
+use crate::pool_fetching::{Pool, PoolFetcher, PoolFetching};
+use crate::Web3;
+use model::TokenPair;
+use std::collections::HashSet;
+use std::sync::Arc;
+use structopt::clap::arg_enum;
+
+arg_enum! {
+    #[derive(Debug)]
+    pub enum PriceEstimationSources {
+        Uniswap,
+        Sushiswap,
+    }
+}
+
+pub struct PoolAggregator {
+    pub pool_fetchers: Vec<PoolFetcher>,
+}
+
+impl PoolAggregator {
+    pub async fn from_sources(
+        sources: Vec<PriceEstimationSources>,
+        chain_id: u64,
+        web3: Web3,
+    ) -> Self {
+        let mut pool_fetchers = vec![];
+        for source in sources {
+            let pair_provider: Arc<dyn AmmPairProvider>;
+            match source {
+                PriceEstimationSources::Uniswap => {
+                    pair_provider = Arc::new(UniswapPairProvider {
+                        factory: contracts::UniswapV2Factory::deployed(&web3)
+                            .await
+                            .expect("couldn't load deployed uniswap router"),
+                        chain_id,
+                    });
+                    tracing::info!("Constructed Uniswap Pair Provider");
```

Comment:
Not sure if these logs and the two that follow are really all that necessary, but they helped me debug, will remove if requested.

Reply:
They only get logged once on startup, but I wouldn't consider them useful in prod. Please remove.